modules/living_spaces_default/living_spaces_default.module

```diff
@@ -85,13 +101,28 @@ function living_spaces_default_query_views_entity_query_alter(SelectInterface $q
  */
 function living_spaces_default_group_operations_alter(array &$operations, GroupInterface $group) {
   if (array_key_exists('group-join', $operations) &&
-    $group->get('is_default')->getString() &&
-    !\Drupal::currentUser()->hasPermission('manage living spaces default content')
+    $group->get('is_default')->getString()
   ) {
     unset($operations['group-join']);
   }
 }
 
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function living_spaces_default_group_relationship_insert(GroupRelationshipInterface $relationship) {
+  if ('c84c2661-b8e4-4759-a479-4b8c91a25dd8' == $relationship->getGroup()->uuid()) {
+    drupal_register_shutdown_function('living_spaces_group_relationship_post_delete', $relationship);
+  }
+}
```

Comment:
Remove this individual hooks and implement one universal that takes care of all spaces which marked as is_default